src/mono/wasm/debugger/BrowserDebugProxy/MemberObjectsExplorer.cs

```diff
@@ -150,8 +152,9 @@ private static string GetNamePrefixForValues(string memberName, string typeName,
                 {
                     // a collection - expose elements to be of array scheme
                     var memberNamedItems = members
-                        .Where(m => m["name"]?.Value<string>() == "Items" || m["name"]?.Value<string>() == "_items")
-                        .FirstOrDefault();
+                        .FirstOrDefault(m => m["name"]?.Value<string>() == "Items" || m["name"]?.Value<string>() == "_items");
```

Comment:
background:
- `List<T>` allocates a private array (`_items`) of `Capacity` size. The size for that can be obtained from `Capacity`, or `_items.Length`.
- It also has a private `int _size`, which has the number of *actual* elements in the list (`== list.Count`). And this changes when you add/remove elements to the list.
- At some threshold, `_items` array is reallocated to have a bigger size, and the elements are copied over.

https://github.com/dotnet/runtime/blob/51e3afb2ba42609c3711cb50fb5eaa6800bc1fde/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs#L25-L26

Based on that:
- `maxSize` is really the size of the internal array
- `_items` here makes sense, since that is a fixed private field in `List<T>` (specifically in `List<T>` though)
- Also, the type has `[DebuggerTypeProxy(typeof(ICollectionDebugView<>))]` which has:

https://github.com/dotnet/runtime/blob/51e3afb2ba42609c3711cb50fb5eaa6800bc1fde/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/ICollectionDebugView.cs#L19-L28

.. which returns an array of size == list.Count .

- So, is this using the type proxy, or are we reading from the actual object? If it's from the former then `Items`, and for the latter `_items` would be relevant. Feel free to correct me, I might be missing something.

Reply:
I don't see any errors in the above. I only cannot understand what mechanism makes the decision if to use `Items` or `_items`. I don't think that it's too important but if we knew the rule it would help in simplifying the code. I think I will add some comments to the current code